### ISSUE TITLE
Linker errors in tests on MacOS 

### ISSUE THREAD

**Issue by jppelteret** [dealii/dealii#4722]
Failing tests:
- lac/constraints_local_to_global_rect
```
1894: Undefined symbols for architecture x86_64:
1894:   "void dealii::ConstraintMatrix::distribute_local_to_global<dealii::FullMatrix<double> >(dealii::FullMatrix<dealii::FullMatrix<double>::value_type> const&, std::__1::vector<unsigned int, std::__1::allocator<unsigned int> > const&, dealii::ConstraintMatrix const&, std::__1::vector<unsigned int, std::__1::allocator<unsigned int> > const&, dealii::FullMatrix<double>&) const", referenced from:
1894:       test() in constraints_local_to_global_rect.cc.o
```
- matrix_free/assemble_matrix_01
```
2394: Undefined symbols for architecture x86_64:
2394:   "void dealii::internal::MatrixFreeFunctions::ShapeInfo<dealii::VectorizedArray<double> >::reinit<2>(dealii::Quadrature<1> const&, dealii::FiniteElement<2, 2> const&, unsigned int)", referenced from:
2394:       dealii::internal::MatrixFreeFunctions::ShapeInfo<dealii::VectorizedArray<double> >::ShapeInfo<2>(dealii::Quadrature<1> const&, dealii::FiniteElement<2, 2> const&, unsigned int) in assemble_matrix_01.cc.o
2394:   "void dealii::internal::MatrixFreeFunctions::ShapeInfo<dealii::VectorizedArray<double> >::reinit<3>(dealii::Quadrature<1> const&, dealii::FiniteElement<3, 3> const&, unsigned int)", referenced from:
2394:       dealii::internal::MatrixFreeFunctions::ShapeInfo<dealii::VectorizedArray<double> >::ShapeInfo<3>(dealii::Quadrature<1> const&, dealii::FiniteElement<3, 3> const&, unsigned int) in assemble_matrix_01.cc.o
```
- matrix_free/assemble_matrix_02: Same as above
- matrix_free/assemble_matrix_03 : Same as above
- matrix_free/matrix_vector_15: Same as above
- matrix_free/matrix_vector_16: Same as above

**Comment by jppelteret** [comment]
Fixed by #4752.

**Comment by masterleinad** [comment]
I am closing since I don't see any problems on `macOS` with the tests mentioned above.
